components/DataSchema.tsx

```diff
@@ -106,10 +90,86 @@ function getColumnDef(dataSchemaMap?: {
+            selector: ColumnSelector.ConditionalIf,
+            cell: (schemaData: DataSchemaData) => {
+                const conditionalIfList = [];
+
+                // Check if this schema is a dependency of any other schema
+                if (dataSchemaMap) {
+                    for (const [key, value] of Object.entries(dataSchemaMap)) {
+                        if (
+                            value.requiredDependencies &&
+                            Array.isArray(value.requiredDependencies)
+                        ) {
+                            const isDependency = value.requiredDependencies.some(
+                                (dep) =>
+                                    (typeof dep === 'string'
+                                        ? dep
+                                        : dep['@id']) === schemaData.id
+                            );
+                            if (isDependency && value.attribute) {
+                                conditionalIfList.push(value.attribute);
+                            }
+                        }
+                    }
+                }
+
+                return (
+                    <ConditionalIfValues
+                        attribute={schemaData.attribute}
+                        attributes={conditionalIfList}
+                    />
+                );
+            },
+            wrap: true,
+            minWidth: '250px',
+            sortable: true,
+        },
+        [ColumnName.DataType]: {
+            name: ColumnName.DataType,
+            selector: ColumnSelector.DataType,
+            wrap: true,
+            sortable: true,
+            format: (schemaData: DataSchemaData) => {
+                if (
+                    schemaData.validationRules &&
+                    Array.isArray(schemaData.validationRules)
+                ) {
+                    const dataType = schemaData.validationRules.find(
+                        (rule) => typeof rule === 'object' && 'type' in rule
+                    );
+
+                    if (
+                        dataType &&
+                        typeof dataType === 'object' &&
+                        'type' in dataType
+                    ) {
+                        switch (dataType) {
+                            case 'int':
+                                return 'Integer';
+                            default:
+                                return 'String';
+                        }
+                    }
+
+                    // If no specific type is found, check if it's an enum (array of allowed values)
+                    if (
+                        schemaData.validationRules.some((rule) =>
+                            Array.isArray(rule)
+                        )
+                    ) {
+                        return 'Enum';
+                    }
+                }
+
+                // Default to String if no validation rules or unrecognized type
+                return 'String';
```

Comment:
This can also be a utility function in `dataShemaHelpers`. Something like `getDataType`. We can also define an enum for this. 

```JavaScript
enum SchemaDataType {
    Integer = "Integer",
    String = "String",
    Enum = "Enum"
}